internal/interfaces/grpc/interceptor/interceptor.go

```diff
@@ -2,23 +2,33 @@ package interceptor
 
 import (
 	middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/tdex-network/tdex-daemon/config"
 	dbbadger "github.com/tdex-network/tdex-daemon/internal/infrastructure/storage/db/badger"
+	"github.com/tdex-network/tdex-daemon/pkg/macaroons"
 	"google.golang.org/grpc"
 )
 
 // UnaryInterceptor returns the unary interceptor
-func UnaryInterceptor(dbManager *dbbadger.DbManager) grpc.ServerOption {
+func UnaryInterceptor(
+	dbManager *dbbadger.DbManager,
+	macaroonService macaroons.Service,
```

Comment:
it would be better to pass the pointer to the macaroon service so there's no need to dereference it elsewhere.